index.html

```diff
@@ -4564,35 +4572,43 @@ <h3>Verification</h3>
 
         <ol class="algorithm">
           <li>
-Set <var>result</var> to an empty [=map=].
-          </li>
-          <li>
 Ensure that the securing mechanism has properly protected the
 <a>conforming document</a> by performing the following steps:
             <ol class="algorithm">
-              <li>
-Set <var>securingMechanismVerified</var> to the result of executing the
-<a href="#verify-securing-mechanism">securing mechanism verification
-algorithm</a> with <var>inputBytes</var> and <var>mediaType</var> passed as
-inputs to the algorithm.
+             <li>
+Set the |verifyProof| function by using the |inputMediaType| and the
+<a data-cite="?VC-SPECS#securing-mechanisms">Securing Mechanisms</a> section of
+the [[[?VC-SPECS]]] [[?VC-SPECS]], or other mechanisms known to the
+implementation, to determine the cryptographic suite to use when verifying
+the securing mechanism. The |verifyProof| function MUST implement the interface
+described in <a href="#securing-mechanisms"></a>.
+                <div class="issue"
+                     title="Mechanism for 'determining' is being detailed">
+At present, the Working Group is concerned that the algorithm for "determining"
+might need to be more formally defined. At present, no implementation has
+had an issue determining the proper |verifyProof| algorithm to use, but the
+Working Group is attempting to see if saying more here would be worthwhile.
+                </div>
               </li>
               <li>
-Set <var>status</var>, <var>document</var>, <var>mediaType</var>,
-<var>warnings</var>, and <var>errors</var> in <var>result</var> to their
-respective properties in <var>securingMechanismVerified</var>.
+Set <var>result</var> to the result of passing <var>inputBytes</var> and
```

Comment:
This algorithm could be written to receive a document instead of bytes (i.e., the bytes have already been parsed) -- and we should allow this. I suspect most implementations already work this way and are built to run on top of a pipeline that has already parsed the bytes as JSON.

Reply:
There is a PR this this PR is built on top of that does this: https://github.com/w3c/vc-data-model/pull/1393/commits/02c3744873614afe5ab0ae0cc14d5b0918c15916

I expect that PR to be merged and this PR will inherit those changes that allow bytes or a document. I believe it has to be both because of enveloping data formats, which are often sequences of bytes that don't have a document representation that are passed to the verification algorithm.